Ical.Net/DefaultTimeZoneResolver.cs

```diff
@@ -0,0 +1,96 @@
+﻿//
+// Copyright ical.net project maintainers and contributors.
+// Licensed under the MIT license.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using NodaTime;
+using NodaTime.TimeZones;
+
+namespace Ical.Net;
+
+public static class DefaultTimeZoneResolver
+{
+    private static Dictionary<string, string> InitializeWindowsMappings()
+        => TzdbDateTimeZoneSource.Default.WindowsMapping.PrimaryMapping
+        .ToDictionary(k => k.Key, v => v.Value, StringComparer.OrdinalIgnoreCase);
+
+    private static readonly Lazy<Dictionary<string, string>> _windowsMapping
+        = new Lazy<Dictionary<string, string>>(InitializeWindowsMappings, LazyThreadSafetyMode.PublicationOnly);
+
+    /// <summary>
+    /// Use this method to turn a raw string into a NodaTime DateTimeZone. It searches all time zone providers (IANA, BCL, serialization, etc) to see if
+    /// the string matches. If it doesn't, it walks each provider, and checks to see if the time zone the provider knows about is contained within the
+    /// target time zone string. Some older icalendar programs would generate nonstandard time zone strings, and this secondary check works around
+    /// that.
```

Comment:
👍

Reply:
The code in this method is copied (almost) 1:1 from the original in `DateUtil`.